dimod/reference/composites/scalecomposite.py

```diff
@@ -135,7 +135,7 @@ def sample(self, bqm, scalar=None, bias_range=1, quadratic_range=None,
 
         sampleset = self.child.sample(bqm, **parameters)
 
-        yield  # wait until sampleset is read for the remaining part
+        yield sampleset  # so that SampleSet.done() works
```

Comment:
How about adding a regression test for this?

Reply:
As far as I know the only place we actually used this (so far) was in `ScaleComposite`. Also, the previous behaviour was explicitly to ignore whatever was yielded first. I do have an integration test ready for dwave-system that I'll add once this is deployed.